core/azure-core/src/main/java/com/azure/core/implementation/Validator.java

```diff
@@ -40,16 +40,7 @@ public static void validate(Object parameter) {
         }
 
         Class<?> type = parameter.getClass();
-        if (type == Double.class
-                || type == Float.class
-                || type == Long.class
-                || type == Integer.class
-                || type == Short.class
-                || type == Character.class
-                || type == Byte.class
-                || type == Boolean.class) {
-            type = wrapperToPrimitive(type);
-        }
```

Comment:
Why did you remove this? Are you sure it is never used?

Reply:
Yeah, the `wrapperToPrimitive` method did nothing but return the input `type`. Not sure why this was added but just to be sure, I added a test case to cover primitive types.

Reply:
I don't believe this is accurate - wrapperToPrimitive returns, e.g. Integer.TYPE when given an Integer.class - these are two separate class types (the first represents the primitive, the second represents the reference type).

Reply:
If the type is Integer.class, the `wrapperToPrimitive` method has a check `if ( !clazz.isPrimitive() ) {return clazz;}` which will just return Integer.class because the `isPrimitive` check fails. So, it will not be able to return `Integer.TYPE`. Maybe that `isPrimitive` check is not required?

Reply:
Yes, good point! I don't know why this code exists. I think it is probably safe to remove it if you've tested it enough.

Reply:
Yeah, I tested with all primitive and boxed types of int, short, long, byte, boolean, char, float and none of them went past the `!class.isPrimitive()` check. So, I think it's safe to remove it.